### PR TITLE
Slice elimination support for opsets 10 and 11

### PR DESCRIPTION
**Pull request by kkaranasos**
**Description**: Updating slice elimination rewrite-rule so that it supports opsets 10 and 11. Starting from opset 10, starts/ends/axes properties of the Slice node are no longer provided as attributes but as inputs to the node. Moreover, there is the newly added property steps (steps should either not exist or be all 1s for Slice to be redundant).

### CODE REVIEW

File: onnxruntime/core/optimizer/initializer.h [hunk 1]
```diff
@@ -268,6 +322,22 @@ class Initializer final {
...
       }
+      case ONNX_NAMESPACE::TensorProto_DataType_INT32: {
+        int32_t* dst = data<int32_t>();
+        const int32_t* src = other.data<int32_t>();
+        for (int i = 0; i < n; i++) {
+          dst[i] += src[i];
+        }
+        break;
+      }
+      case ONNX_NAMESPACE::TensorProto_DataType_INT64: {
```
Comment (skottmckay) [inline onnxruntime/core/optimizer/initializer.h]: We should really just templatize the logic into a small helper function, but that can be done later.

Comment (kkaranasos) [reply onnxruntime/core/optimizer/initializer.h]: yeah I completely agree, just didn't want to introduce more changes right before the release.

File: onnxruntime/core/optimizer/slice_elimination.cc [hunk 2]
```diff
@@ -29,23 +29,94 @@ bool EliminateSlice::SatisfyCondition(const Graph& graph, const Node& node) cons
...
+      return false;
     }
-  } else if (axes.size() != starts.size()) {
-    return false;
   }
 
   // For now eliminate slice operators if starts=0 and ends=MAX_INT.
   // TODO: Take into account the input's shape to get a tighter bound for the ends.
-  for (size_t i = 0; i < axes.size(); ++i) {
+  for (size_t i = 0; i < starts.size(); ++i) {
```
Comment (skottmckay) [inline onnxruntime/core/optimizer/slice_elimination.cc]: Is there an existing bug in that nowhere seems to check the axes values? We can't eliminate the Slice if the axes value is provided but has out of order values can we?

Comment (kkaranasos) [reply onnxruntime/core/optimizer/slice_elimination.cc]: Semantics of slice is not clear to me...
Reading the documentation, I don't understand if Slice is allowed to reorder the axes of the input array. It is allowed to have the axes be out of order, like [1,0] (btw no duplicates are allowed in our shape inference), but that does not mean reordering should happen. Documentation simply says that it is the same as the numpy slice, and in that there is no syntax that allows reordering.

Comment (skottmckay) [reply onnxruntime/core/optimizer/slice_elimination.cc]: I think you're right - it's not for re-ordering so we're good

**Pull request merged**